pkg/rancher-components/src/components/Form/ToggleSwitch/ToggleSwitch.vue

```diff
@@ -27,14 +27,23 @@ export default {
     },
   },
   data() {
-    return { state: this.value === this.onValue };
+    return { state: false };
   },
 
   methods: {
     toggle(neu) {
       this.state = neu === null ? !this.state : neu;
       this.$emit('input', this.state ? this.onValue : this.offValue);
     }
+  },
+
+  watch: {
+    value: {
+      handler() {
+        this.state = this.value === this.onValue;
```

Comment:
These are good questions, but I cannot answer a lot of the `why` as I'm not the original author of this component..  

> Why do we have an `onValue` and `offValue` and why the type is `null` when it is a `Boolean`?

`onValue` and `offValue` seem to allow for other values than `boolean`. I'm guessing the type is `null` when the original author intended to allow for `any`. I see this usage of `null` as a type elsewhere throughout our components. 

> Why is it required to have a watch and a state, something which may be simply set as a prop and updated as emit?

I asked myself the same question but opted to keep the original implementation intact. With this original implementation in place, `state` uses the `value` prop as an initial value and nothing else. `state` is then solely managed internally by the `ToggleSwitch` after this initial. 

> The logic used from the parent should not be passed down to a component, that's against the separations of concerns.

I agree, but I don't understand why you're making this point..

> We may considering removing this logic or create an issue for it while explaining the purpose inside the component.

Much agreed. I think it makes sense to follow up with a refactor of this component. And getting tests in place will help us in this process. Please remember that we are incrementally improving what already exists in production.

Reply:
I think it's also good to mention that we already make use of constructs like `onValue` and `offValue` in `provisioning.cattle.io.cluster`: 

https://github.com/rancher/dashboard/blob/44fd817f5bfbbeac183aaa09e86cdde5b5cdbbec/shell/edit/provisioning.cattle.io.cluster/index.vue#L466-L474

Reply:
Sorry for having so many questions, just I usually tend to do it when it's not documented 😓 
I personally refactor small part of code when I work on it, as that's pretty much the reason we have to extend the code in this case too.
However, it's up to you to make a decision, as it just wanted to be a suggestion for simplify the logic.
Could we add a couple of notes, especially considering it has to be published as library and out of context may be confusing?

Reply:
Since I've seen you added the type, it may probably suffice.

Reply:
No apologies needed. I said it before and I said it again, I think you're asking good questions 😁

There's always a need for documentation where it is lacking, which is unfortunately the case for a lot of our components. I've created #6759 to follow up with some more comments in our code.

There's definitely a strong case for refactoring here and perhaps I should have approached this from the angle of writing tests first and then following up with a separate PR to address behavior. Maybe this would give me more confidence to make a bigger change.. Anyhow, I've created #6760 to circle back and improve the watcher implementation.

Reply:
Fair enough for me. Thanks for writing it down in issues so we don't lose track 👍